d20/tests/test_GameMaster.py

```diff
@@ -318,3 +341,1637 @@ def testhandleAddObject4(self, FileObjectInit):
+    assert gm.extra_players == 'test'
+    assert gm.extra_npcs == 'test'
+    assert gm.extra_backstories == 'test'
+    assert gm.extra_screens == 'test'
+
+    with pytest.raises(TypeError) as excinfo:
+        gm = GameMaster(test='test')
+    assert str(excinfo.value) == "test is an invalid keyword argument"
+
+    del args.temporary
+    with pytest.raises(RuntimeError) as excinfo:
+        gm = GameMaster(options=args)
+    assert str(excinfo.value) == "Expected temporary directory to be" \
+        " specified in options"
+    args.temporary = '/tmp/d20-test'
+
+
+def testBackStoryFactLoad1(monkeypatch):
+    mockBSF = mock.Mock()
+    mock1 = mock.Mock()
+
+    monkeypatch.setattr("d20.Manual.GameMaster.GameMaster.registerBackStories",
+                        mock1)
+    monkeypatch.setattr("d20.Manual.GameMaster.GameMaster.registerNPCs",
+                        mock1)
+    monkeypatch.setattr("d20.Manual.GameMaster.GameMaster.registerPlayers",
+                        mock1)
+    monkeypatch.setattr("d20.Manual.GameMaster.GameMaster.registerScreens",
+                        mock1)
+    monkeypatch.setattr("d20.Manual.GameMaster.resolveBackStoryFacts",
+                        mockBSF)
+
+    args = args_ex
+    args.backstory_facts = """
+    facts:
+        - name: BulkAnalyzeFact
+          arguments:
+            directory: /path/to/files
+            recursive: False
+            enable: True
+        - name: VTDownloadFact
+          arguments:
+            vt_api_key: your_api_key
+            filehash: hash_to_lookup_and_download_for_analysis
+            enable: False
+    """
+
+    gm = GameMaster(options=args)
+    mockBSF.assert_called()
+    gm.cleanup()
+
+
+def testBackStoryFactPathLoad2(monkeypatch):
+    mockBSF = mock.Mock()
+    mock1 = mock.Mock()
+
+    monkeypatch.setattr("d20.Manual.GameMaster.GameMaster.registerBackStories",
+                        mock1)
+    monkeypatch.setattr("d20.Manual.GameMaster.GameMaster.registerNPCs",
+                        mock1)
+    monkeypatch.setattr("d20.Manual.GameMaster.GameMaster.registerPlayers",
+                        mock1)
+    monkeypatch.setattr("d20.Manual.GameMaster.GameMaster.registerScreens",
+                        mock1)
+    monkeypatch.setattr("d20.Manual.GameMaster.resolveBackStoryFacts",
+                        mockBSF)
+
+    tf = tempfile.NamedTemporaryFile(delete=False)
+    tf.write(b"""
+        facts:
+            - name: BulkAnalyzeFact
+              arguments:
+                directory: /path/to/files
+                recursive: False
+                enable: True
+            - name: VTDownloadFact
+              arguments:
+                vt_api_key: your_api_key
+                filehash: hash_to_lookup_and_download_for_analysis
+                enable: False
+    """)
+
+    tf.close()
+    args = args_ex
+    args.backstory_facts_path = tf.name
+
+    GameMaster(options=args)
+    mockBSF.assert_called()
```

Comment:
would it be better to use `assert_called_with(fact abject)` here to make sure it fell into the correct call and got the proper loaded backstory object?